src/praisonai/praisonai/persistence/conversation/_sql_base.py

```
@@ -0,0 +1,459 @@
+                tool_call_id=row['tool_call_id'],
+                metadata=json.loads(row['metadata']) if row['metadata'] else None,
+                created_at=row['created_at']
+            )
+            for row in rows
+        ]
+    
+    def delete_messages(self, session_id: str, message_ids: Optional[List[str]] = None) -> int:
+        """Delete messages. If message_ids is None, delete all messages in session."""
+        def delete_message_rows(conn):
+            if message_ids is None:
+                sql = f"DELETE FROM {self.messages_table} WHERE session_id = {self._param}"
```

Ruff (0.15.15)

[error] Possible SQL injection vector through string-based query construction

(S608)